src/common/types.ts

```diff
@@ -1,15 +1,21 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-import { AwsCredentialIdentity, Provider, RequestSigner } from "@aws-sdk/types";
+import {
+  AwsCredentialIdentity,
+  AwsCredentialIdentityProvider as _AwsCredentialIdentityProvider,
+  RequestSigner,
+} from "@aws-sdk/types";
+
+export type AwsCredentialIdentityProvider = _AwsCredentialIdentityProvider;
 
 export interface MapAuthenticationOptions {
   transformRequest: (url: string, resourceType?: string) => { url: string };
 }
 
 export interface LocationClientConfig {
   signer?: RequestSigner;
-  credentials?: Provider<AwsCredentialIdentity>;
+  credentials?: AwsCredentialIdentityProvider;
```

Comment:
Is `AwsCredentialIdentityProvider` compatible with our old `Provider<AwsCredentialIdentity>`?